src/jit/morph.cpp

```diff
@@ -8109,22 +8109,23 @@ void Compiler::fgMorphRecursiveFastTailCallIntoLoop(BasicBlock* block, GenTreeCa
 
     // If compInitMem is set, we may need to zero-initialize some locals. Normally it's done in the prolog
     // but this loop can't include the prolog. Since we don't have liveness information, we insert zero-initialization
-    // for all non-parameter non-temp locals as well as temp structs with GC fields.
+    // for all non-parameter IL locals as well as temp structs with GC fields.
     // Liveness phase will remove unnecessary initializations.
     if (info.compInitMem)
     {
         unsigned   varNum;
         LclVarDsc* varDsc;
         for (varNum = 0, varDsc = lvaTable; varNum < lvaCount; varNum++, varDsc++)
         {
-            var_types lclType = varDsc->TypeGet();
             if (!varDsc->lvIsParam)
             {
-                if (!varDsc->lvIsTemp || ((lclType == TYP_STRUCT) && (varDsc->lvStructGcCount > 0)))
+                var_types lclType            = varDsc->TypeGet();
+                bool      isILTemp           = (varNum < info.compLocalsCount);
```

Comment:
Shouldn't this variable be named `isNotILTemp` ?

Reply:
If we call local variables from C#(or IL) as temp IL variables, then it should be `isILTemp`.
`lvaTable` starts with IL method arguments, that we skip, than IL temps, that we have to zero-initialize.

Reply:
This name is confusing. I would prefer isUserLocal.

Reply:
Fixed.